src/lib/components/Table.svelte

```diff
@@ -0,0 +1,61 @@
+<script lang="ts">
+	import type { Highscore_type } from '$lib/types/types';
+	import Crown from '$lib/components/icons/Crown.svelte';
+	import { onMount } from 'svelte';
+
+	let {
+		data
+	}: {
+
+		data: Highscore_type[];
+	} = $props();
+
+
+	const data_out: { rank: number; name: string, points: number, highlight: boolean }[] = $state([]);
```

Comment:
Es würde grundsätzlich mehr Sinn machen nur Daten zu übergeben, die dann auch in der Komponenten genutzt werden und die ifs in onMount auszulagern. Oder man sagt, dass die Komponente immer gleich viele Plätze anzeigen soll, dann muss data aber nicht als props übergeben werden sondern kann direkt in der Komponente geladen werden.